### PR TITLE
[RNMobile] File Block - Add Link To functionality

### PR DESCRIPTION
**Pull request by jd-alexander**
Gutenberg Mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/2810
WP Android PR https://github.com/wordpress-mobile/WordPress-Android/pull/13405

## Description

This PR adds the `Link To` functionality in the File Block settings. When you select this option you will be greeted with a modal that allows you to select either an Attachment File or Media File as the link of the File Block. Doing so will add the appropriate `<a href>` tag to the block. 

## How has this been tested?

To properly test this behavior you will need to use one of the parent apps as the `getMedia` function requires the file to exist within the site. So if you test this PR here, you will see the URL as the only Link To option as seen below.

<kbd></kbd>

Once you have this PR's submodule reference updated in the respective parent app project, you can follow the testing instructions below. 

#### Media File

1. Create a new post. 
2. Add a File Block. 
3. Add a file. 
4. Click the settings button. 
5. Ensure the Link To is set to Media File. 
6. Close this modal. 
7. Switch to HTML mode. 
8. Go to the File Block within the html.
9. Observe that the href within the first a tag has a link to the media file. To be sure, you will see the file's extension here.
 
<kbd></kbd>

#### Attachment Page

1. Create a new post. 
2. Add a File Block. 
3. Add a file. 
4. Click the settings button. 
5. Ensure the Link To is set to Attachment Page. 
6. Close this modal. 
7. Switch to HTML mode. 
8. Go to the File Block within the html.
9. Observe that the the href within the first a tag has a link to a WordPress page. 

<kbd></kbd>

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/file/edit.native.js [hunk 1]
```diff
@@ -326,12 +363,21 @@ export class FileEdit extends Component {
...
 			},
 			createInfoNotice,
 		};
 	} ),
+	withSelect( ( select, props ) => {
+		const { attributes } = props;
+		const { id } = attributes;
+		return {
+			media:
+				id === undefined ? undefined : select( 'core' ).getMedia( id ),
```
Comment (jd-alexander) [inline packages/block-library/src/file/edit.native.js:379]: @etoledom will this `getMedia` call work if the upload wasn't actually done already? I remember seeing a comment about the current implementation being a fake upload hence my question.

Comment (etoledom) [reply packages/block-library/src/file/edit.native.js:379]: As we talked internally, is true that we are faking the upload on the demo apps, but on the parent apps is all working as expected, so this feature is best to be tested with the parent apps.

Even though, even with the fake upload, I believe we are sending all needed information back to gutenberg for it to create the appropriate media object, so this is worth checking too 👍

Comment (jd-alexander) [reply packages/block-library/src/file/edit.native.js:379]: > On a second thought, even with the fake upload, I believe we are sending all needed information back to gutenberg for it to create the appropriate media object, so this is worth checking too 👍

Indeed! This is what I was wondering as well. I will take another look. 🙏

**Comment by jd-alexander** [comment]
An alternative solution that utilizes the sidebar functionality to open and close the settings of a block was used instead of duplicating the logic. This solution was implemented in https://github.com/WordPress/gutenberg/pull/27234

**Pull request closed without merging**